functests/utils/discovery/discovery.go

```diff
@@ -1,12 +1,58 @@
 package discovery
 
 import (
+	"context"
 	"os"
 	"strconv"
+
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	performancev1alpha1 "github.com/openshift-kni/performance-addon-operators/pkg/apis/performance/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // Enabled indicates whether test discovery mode is enabled.
 func Enabled() bool {
 	discoveryMode, _ := strconv.ParseBool(os.Getenv("DISCOVERY_MODE"))
 	return discoveryMode
 }
+
+// GetDiscoveryPerformanceProfile returns an existing profile in the cluster with the most nodes using it.
+// In case no profile exists - return nil
+func GetDiscoveryPerformanceProfile(exclude ...string) (*performancev1alpha1.PerformanceProfile, error) {
+	performanceProfiles, err := profiles.GetAllProfiles()
+	if err != nil {
+		return nil, err
```

Comment:
Test wise, would it return a "nil"  error? Would we need to go to the code to see were it failed or would we receive a nice "Error fetching profiles", and in this case, would we have some info on "why"? (is the operator healthy?)

Reply:
this is the way we move errors up the stack in case we encounter them (kindof throwable) a nicer way would be to wrap this error with another message and throw a new error but i think its redundant here and you will get the full path of the error here in case you want to debug the test code. 
There should be more places in code like this, we can consider a total refactor of all if you see fit